resolwe_bio/processes/import_data/basespace.yml

```diff
@@ -0,0 +1,55 @@
+- slug: basespace-file-import
+  name: BaseSpace File Import
+  requirements:
+    expression-engine: jinja
+    executor:
+      docker:
+        image: resolwebio/utils:1.2.0
+    resources:
+      network: true
+      secrets: true
+  data_name: 'BaseSpace File Import ({{ file_id|default("?") }})'
+  version: 0.0.1
+  type: data:file
+  category: upload
+  persistence: RAW
+  description: >
+    Import file from Illumina BaseSpace.
+  input:
+    - name: file_id
+      label: BaseSpace file ID
+      type: basic:string
+    - name: access_token_secret
+      label: BaseSpace access token
+      type: basic:secret
+      description: >
+        BaseSpace access token secret handle needed to download the file.
+    - name: verbose
+      label: Verbose
+      description: >
+        Print additional information when errors occur.
+      type: basic:boolean
+      default: false
+  output:
+    - name: file
+      label: File
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      {% set SECRET_PATH = '/secrets/' ~ access_token_secret.handle %}
+
+      BASESPACE_OUTPUT=$(basespace_download.py \
+        --file-id {{ file_id }} \
+        --access-token-secret-path {{ SECRET_PATH }} \
+        {% if verbose %} --verbose {% endif %} \
+      )
+      re-checkrc "Failed downloading files from BaseSpace."
+
+      function extract_filename_from_output() {
```

Comment:
Why can't `basespace_download.py` also handle this? It seems strange to me that you need to do post-processing in bash for a script that you provide.

Reply:
Perhaps this could be an output format parameter, like `--output filename` would output just the filename while `--output full` would output everything as it is now.

Reply:
This is used to obtain the filename which is needed in `re-save-file`.

Is there any other way of extracting "return" values from python scripts?

Reply:
This wouldn't be a problem if processes were written in Python.

Reply:
> This wouldn't be a problem if processes were written in Python.

Sure, but until they are I think the `--output` flag is not that bad.

Reply:
The script can take multiple files on the input and prints downloaded filenames in the form of `filename=foo.fastq.gz`, so by `--output filename` you mean only `filename=...` would be printed?

The thing is that `print('filename=` is the only print statement, unless an error happens in which case filename is not used. Therefore having `--output full` does the same thing as `--output filename`.

Reply:
Output with `--output full`:
```
filename=bar
somethingelse=foo
```

Output with `--output filename`:
```
bar
```

The point is that you make the output such that you don't need any post-processing.

Reply:
BTW, if the filename is the only thing printed, why doesn't the process just output filenames separated by newlines instead of printing them in a `key=value` format?

Reply:
To make it more robust if anything changes in the script (e.g. someone adds print statement and breaks processes). I like the idea of `key=value` to be honest and don't mind doing a little post processing.

Reply:
In this case, I think that supporting a flag like I mentioned above should be the right way. E.g. by default or if `--output full` is specified, you keep the `key=value` format, but if one specifies `--output filename` (or other keys in the future) only the values for that key are printed. In this way, you don't need any post-processing.

Reply:
Fixed.